azure/compute/vm/azurevm.go

```diff
@@ -0,0 +1,60 @@
+package vm
+
+import (
+	"errors"
+
+	"github.com/DataDog/test-infra-definitions/azure"
+	"github.com/DataDog/test-infra-definitions/azure/compute"
+	"github.com/DataDog/test-infra-definitions/command"
+	"github.com/DataDog/test-infra-definitions/common/os"
+	"github.com/DataDog/test-infra-definitions/common/vm"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+type AzureVM struct {
+	runner *command.Runner
+}
+
+// NewAzureVM creates a new azure instance. By default use WithOS(os.UbuntuOS, os.AMD64Arch).
+func NewAzureVM(ctx *pulumi.Context, options ...func(*Params) error) (*AzureVM, error) {
+	env, err := azure.AzureEnvironment(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	params, err := newParams(env, options...)
+	if err != nil {
+		return nil, err
+	}
+
+	if params.common.OS.GetOSType() != os.UbuntuOS {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Should we instead comment out `windows` in `azure/compute/os/os.go` in ` GetOSes` while we don' support it ?

Reply:
I am working on fixing the issue (AETT-43).